pkg/controllers/management/usercontrollers/controller.go

```diff
@@ -317,6 +347,75 @@ func (c *ClusterLifecycleCleanup) updateClusterRoleBindingOwner(
 	})
 }
 
+func cleanupNamespaces(client kubernetes.Interface) error {
+	logrus.Debug("Starting cleanup of local cluster namespaces")
+	namespaces, err := client.CoreV1().Namespaces().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range namespaces.Items {
+		err = tryUpdate(func() error {
+			nameSpace, err := client.CoreV1().Namespaces().Get(ns.Name, metav1.GetOptions{})
+			if err != nil {
+				if apierror.IsNotFound(err) {
+					return nil
+				}
+				return err
+			}
+
+			var updated bool
+
+			// Cleanup finalizers
+			if len(nameSpace.Finalizers) > 0 {
+				finalizers := []string{}
+				for _, finalizer := range nameSpace.Finalizers {
```

Comment:
i think `range nameSpace.Finalizers` should take care of the zero length case.

Reply:
I added the check to stop the allocation and the check on 377 if there aren't any